engine/resources/file.go

```diff
@@ -137,9 +137,19 @@ func (obj *FileRes) isDir() bool {
 // the case where the mode is not specified. The caller should check obj.Mode is
 // not empty.
 func (obj *FileRes) mode() (os.FileMode, error) {
-	m, err := strconv.ParseInt(obj.Mode, 8, 32)
+	n, err := strconv.ParseInt(obj.Mode, 8, 32)
+	m := os.FileMode(n)
```

Comment:
Having this between the `xxx, err := ...` and the `if err != nil` is awkward, might be better to inline the code and exit early if `err == nil`.